sympy/functions/combinatorial/factorials.py

```diff
@@ -1154,29 +1154,27 @@ def _eval_rewrite_as_Piecewise(self, n, k, **kwargs):
 
     def _eval_rewrite_as_factorial(self, n, k, **kwargs):
         from sympy import Pow
-        isint = lambda x: x.is_integer is True
-        nneg = lambda x: x.is_nonnegative is True
-        neg = lambda x: x.is_negative is True
-        if And(isint(n), isint(k)):
-            if And(neg(n), nneg(k)):
-                return(Pow(-1, k)*factorial(k - n - 1)/(factorial(k)*factorial(-n - 1)))
-            if And(neg(n), neg(k)):
-                return (Pow(-1, n - k)*factorial(-k - 1)/(
-                    factorial(n - k)*factorial(-n - 1)))
-            if And(nneg(n), neg(k)):
-                return S.Zero
-            if And(nneg(n), nneg(k)):
-                return factorial(n)/(factorial(k)*factorial(n - k))
+
+        if fuzzy_and([n.is_integer, k.is_integer]):
+            if n.is_negative:
+                if k.is_nonnegative:
+                    return(Pow(-1, k)*factorial(k - n - 1)/(
+                        factorial(k)*factorial(-n - 1)))
+                if k.is_negative:
+                    return (Pow(-1, n - k)*factorial(-k - 1)/(
+                        factorial(n - k)*factorial(-n - 1)))
+            elif n.is_nonnegative:
+                if k.is_negative:
+                    return S.Zero
+                if k.is_nonnegative:
+                    return factorial(n)/(factorial(k)*factorial(n - k))
+        if fuzzy_and([n.is_integer, k.is_integer]) is False:
+            return factorial(n)/(factorial(k)*factorial(n - k))
 
     def _eval_rewrite_as_gamma(self, n, k, **kwargs):
         from sympy import gamma, Pow
-        ki = k.is_integer
-        nki = (n - k).is_integer
-        if ki is None:
-            return
-        if nki is None:
-            return
-        if n.is_integer and k.is_integer:
+
+        if fuzzy_and([n.is_integer, k.is_integer]):
```

Comment:
If you are only going to act on the True result then the `and` version is as direct as you can get. This change does the same thing but adds another function call.

Reply:
Ahhh yeah 😄, I'll change that